kits/arm/arm.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+import hebi
+from time import sleep, time, perf_counter
+import numpy as np
+
+import os
+import sys
+
+# ------------------------------------------------------------------------------
+# Add the root folder of the repository to the search path for modules
+root_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))
+sys.path = [root_path] + sys.path
+# ------------------------------------------------------------------------------
+
+from util.math_utils import get_grav_comp_efforts
+from util.math_utils import gravity_from_quaternion
+
+class ArmParams(object):
+    
+    def __init__(self, family, moduleNames, hrdf):
+        self.family = family
+        self.hrdf = hrdf
+        self.moduleNames = moduleNames
+
+
+    
+class Arm():
+    
+    def __init__(self, params):
+        # Create arm from lookup
+        self.lookup = hebi.Lookup()
+        sleep(2)
+        print(self.lookup)
+        self.grp = self.lookup.get_group_from_names([params.family], params.moduleNames)
+        print(self.grp)
+        
+        # Create robot model
+        try:
+            self.model = hebi.robot_model.import_from_hrdf(params.hrdf)
+        except Exception as e:
+            print('Could not load HRDF: {0}'.format(e))
+            sys.exit()
+        
+        # Create comand and feedback
+        self.cmd = hebi.GroupCommand(self.grp.size)
+        self.fbk = hebi.GroupFeedback(self.grp.size)
+        self.prev_fbk = self.fbk
+        self.grp.get_next_feedback(reuse_fbk=self.fbk)
+        
+        # Zero our initial comand setting vars
+        self.pos_cmd = self.fbk.position
+        self.vel_cmd = np.zeros(self.grp.size)
+        self.accel_cmd = np.zeros(self.grp.size)
+        self.eff_cmd = np.zeros(self.grp.size)
+        
+        # Setup gravity vector for grav comp
+        self.gravity_vec = [0, 0, 1]
+        gravity_from_quaternion(self.fbk.orientation[0], output=self.gravity_vec)
+        self.gravity_vec = np.array(self.gravity_vec)
+        
+        # Zero times
+        self.time_now = time()
+        self.traj_start_time = self.time_now
+        self.duration = 0
+        self.t_traj = 0
+        
+        self.set_goal_first = True
+        self.at_goal = True
+        self.goal = "grav comp"
+        
+    
+    # Loads gains from given gains file    
+    def loadGains(self, gains_file):
+        try:
+            self.cmd.read_gains(gains_file)
+            if not self.grp.send_command_with_acknowledgement(self.cmd):
+                raise RuntimeError('Did not receive ack from group.')
+            print('Successfully read gains from file and sent to module.')
+        except Exception as e:
+            print('Problem reading gains from file or sending to module: {0}'.format(e))
+    
+    
+    # Updates feedback and generates the basic command for this timestep
+    def update(self):
+        self.prev_fbk = list(self.fbk)
```

Comment:
Why is this a list?

Reply:
I'm responsible for this. This clones the IOState. Before this was
```
self.prev_fbk = self.fbk
```
I guess the less cryptic way to do this would be
```
self.prev_list = self.fbk.copy()
```

This wasn't a thing until Python3.3, so I remembered the list(list) way instead

Reply:
Is [`feedback.copy_from`](http://docs.hebi.us/docs/python/2.1.0/hebi/py_api.html#hebi.GroupFeedback.copy_from) the behavior you want here?